### PR TITLE
Add system.mem.slab_reclaimable gauge

### PR DESCRIPTION
**Pull request by jared-gs**
### What does this PR do?

* Adds a gauge for `system.mem.slab_reclaimable`. This is part of slab memory that might be reclaimed (i.e. caches).

### Motivation

Datadog 7.x adds `SReclaimable` memory, if available on the system, to the `system.mem.cached` gauge by default: https://github.com/shirou/gopsutil/commit/f9e238c38b5f16a36794dd2f0f751c7376c64a60.

This may lead to inconsistent metrics for clients migrating from Datadog 5.x, where `system.mem.cached` didn't include `SReclaimable` memory. Adding a gauge for `system.mem.slab_reclaimable` allows inverse calculation to remove this value from the `system.mem.cached` gauge.

### Additional Notes

This PR is a follow up to https://github.com/DataDog/datadog-agent/issues/5038.

### CODE REVIEW

File: pkg/collector/corechecks/system/memory_nix_test.go [hunk 1]
```diff
@@ -70,6 +71,7 @@ func TestMemoryCheckLinux(t *testing.T) {
 	mock.On("Gauge", "system.mem.buffered", 353818902528.0/mbSize, "", []string(nil)).Return().Times(1)
 	mock.On("Gauge", "system.mem.shared", 327680000000.0/mbSize, "", []string(nil)).Return().Times(1)
 	mock.On("Gauge", "system.mem.slab", 327680000000.0/mbSize, "", []string(nil)).Return().Times(1)
+	mock.On("Gauge", "system.mem.sreclaimable", 327680000000.0/mbSize, "", []string(nil)).Return().Times(1)
```
Comment (mx-psi) [inline pkg/collector/corechecks/system/memory_nix_test.go:74]:
```suggestion
	mock.On("Gauge", "system.mem.slab_reclaimable", 327680000000.0/mbSize, "", []string(nil)).Return().Times(1)
```

File: pkg/collector/corechecks/system/memory_nix_test.go [hunk 2]
```diff
@@ -178,6 +180,7 @@ func TestSwapMemoryError(t *testing.T) {
...
```
Comment (mx-psi) [inline pkg/collector/corechecks/system/memory_nix_test.go:183]:
```suggestion
	mock.On("Gauge", "system.mem.slab_reclaimable", 327680000000.0/mbSize, "", []string(nil)).Return().Times(1)
```

File: pkg/collector/corechecks/system/memory_nix.go [hunk 3]
```diff
@@ -89,6 +89,7 @@ func (c *MemoryCheck) linuxSpecificVirtualMemoryCheck(v *mem.VirtualMemoryStat)
 	sender.Gauge("system.mem.buffered", float64(v.Buffers)/mbSize, "", nil)
 	sender.Gauge("system.mem.shared", float64(v.Shared)/mbSize, "", nil)
 	sender.Gauge("system.mem.slab", float64(v.Slab)/mbSize, "", nil)
+	sender.Gauge("system.mem.sreclaimable", float64(v.SReclaimable)/mbSize, "", nil)
```
Comment (mx-psi) [inline pkg/collector/corechecks/system/memory_nix.go:92]:
```suggestion
	sender.Gauge("system.mem.slab_reclaimable", float64(v.SReclaimable)/mbSize, "", nil)
```

**Review by mx-psi — CHANGES_REQUESTED** [review]
Thanks for the PR @jared-gs! It looks good to me overall, I suggested renaming the metric so that it is easier to understand.

Apart from the suggested changes, in order for us to accept your PR we need you to:

- Sign the Datadog [CLA](https://cla.datadoghq.com/DataDog/datadog-agent?pullRequest=6053). The email account associated to your commits needs to be added to the Github account you use to sign the commits.
- Add a brief note with the changes this PR adds. To do this [follow these instructions](https://github.com/DataDog/datadog-agent/blob/master/docs/dev/contributing.md#reno).

File: pkg/collector/corechecks/system/memory_nix.go (hunk 3 above)
Comment (jared-gs) [reply pkg/collector/corechecks/system/memory_nix.go]: Updated!

File: pkg/collector/corechecks/system/memory_nix_test.go (hunk 2 above)
Comment (jared-gs) [reply pkg/collector/corechecks/system/memory_nix_test.go]: Updated!

File: pkg/collector/corechecks/system/memory_nix_test.go (hunk 1 above)
Comment (jared-gs) [reply pkg/collector/corechecks/system/memory_nix_test.go]: Updated!

**Comment by jared-gs** [comment]
Thanks @mx-psi,

I've incorporated those suggestions.

**Review by mx-psi — APPROVED** [review]
LGTM! I will merge it once the CLA check passes (you can see the email account that you used for the commits on https://github.com/DataDog/datadog-agent/pull/6053.patch)

**Comment by jared-gs** [comment]
Thank you so much for your patience as I figured this out! 😅 

I think the CLA is good now.

**Comment by mx-psi** [comment]
No probs, thanks again!

**Pull request merged**